### ISSUE TITLE
Crash when restoring session after Windows/PC boot

### ISSUE THREAD

**Issue by Helldembez** [brave/brave-browser#4764]
## Description 
When starting Brave after booting the pc the previous session crashes and closes Brave. When starting Brave after the crash it doesn't crash again just lost its session. Happens every time after reboot. This is on a fresh windows 10 install and fresh brave install. Happened before fresh install too.

## Steps to Reproduce

   1. Start Brave and browse a bit.
   2. Turn off PC(With brave open)
   3. Boot Pc
   4. Start Brave -> Crash

## Actual result:
Lost session when starting brave after brave crash.

## Expected result:
Expecting my previous session without a crash.

## Reproduces how often: 
So far I've reproduced this multiple times. Basically every time boot my pc.

## Brave version (brave://version info)
Brave | 0.65.118 Chromium: 75.0.3770.80 (Official Build) (64-bit)
-- | --
Revision | 9a9aa15057b6b2cc0909bdcf638c0b65ecd516f2-refs/branch-heads/3770@{#948}
OS | Windows 10 OS Version 1809 (Build 17763.529)

## Version/Channel Information:

- Can you reproduce this issue with the current release?
Yes
- Can you reproduce this issue with the beta channel?
Not tested
- Can you reproduce this issue with the dev channel?
Not tested
- Can you reproduce this issue with the nightly channel?
Not tested

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
Not tested
- Does the issue resolve itself when disabling Brave Rewards?
Not tested
- Is the issue reproducible on the latest version of Chrome? 
No

## Miscellaneous Information:
2 Crash report ids since fresh windows install.
4073d1d1b8f0aca1 
97f08b5aaf72136e

**Comment by Helldembez** [comment]
Update: Only seems to happen when shutting down while browser open.

**Comment by chrisknepper** [comment]
Experiencing the same thing, only happens on Windows.

**Comment by rebron** [comment]
cc: @simonhong Can you take a look?

**Comment by simonhong** [comment]
```
Crash reason:  EXCEPTION_ACCESS_VIOLATION_READ
Crash address: 0x1fdea2a2d6a
Process uptime: 2 seconds

Thread 9 (crashed)
 0  chrome.dll!memcmp [memcmp.asm : 72 + 0x0]
    rax = 0x0000000000000000   rdx = 0xfffffedcd315c8d2
    rcx = 0x000001fdea2a2d6a   rbx = 0x000001fdea9b6810
    rsi = 0x000000dabd3ff288   rdi = 0x0000000000000000
    rbp = 0x000000000000002f   rsp = 0x000000dabd3ff1e8
     r8 = 0x000000000000000d    r9 = 0x000001fdecd7e201
    r10 = 0x000000000000000d   r11 = 0x000000000000000d
    r12 = 0x000000dabd3ff288   r13 = 0x000000dabd3ff518
    r14 = 0x000001fdea9ab270   r15 = 0x000000dabd3ff649
    rip = 0x000001fde077aeb0
    Found by: given as instruction pointer in context
 1  chrome.dll!NoFingerprintDomain::operator==(NoFingerprintDomain const &) [no_fingerprint_domain.cc : 102 + 0xd]
    rsp = 0x000000dabd3ff1f0   rip = 0x000001fdde9a82c6
    Found by: stack scanning
 2  chrome.dll!HashSet::Find(NoFingerprintDomain const &) [hash_set.h : 145 + 0xc]
    rsp = 0x000000dabd3ff220   rip = 0x000001fdde9a34a5
    Found by: stack scanning
 3  chrome.dll!isNoFingerprintDomainHashSetMiss(HashSet *,char const *,int) [ad_block_client.cc : 685 + 0x19]
    rsp = 0x000000dabd3ff260   rip = 0x000001fdde9a3409
    Found by: stack scanning
 4  chrome.dll!operator new(unsigned __int64) [new_scalar.cpp : 35 + 0x8]
    rsp = 0x000000dabd3ff270   rip = 0x000001fde0776fff
    Found by: stack scanning
 5  chrome.dll!AdBlockClient::matches(char const *,FilterOption,char const *,Filter * *,Filter * *) [ad_block_client.cc : 871 + 0x12]
    rsp = 0x000000dabd3ff2e0   rip = 0x000001fdde9a3cc1
    Found by: stack scanning
```
```
Crash reason:  EXCEPTION_ACCESS_VIOLATION_READ
Crash address: 0x8
Process uptime: 1 seconds

Thread 35 (crashed)
 0  chrome.dll!Filter::matchesOptions(char const *,FilterOption,char const *) [filter.cc : 436 + 0x0]
    rax = 0x0000000000004080   rdx = 0x00000151c822b9c0
    rcx = 0x0000000000000000   rbx = 0x0000000000000000
    rsi = 0x00000151c822b9c0   rdi = 0x00000151c822b9c8
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff090
     r8 = 0x0000000000004080    r9 = 0x000000382bdff5a8
    r10 = 0x0000000000000001   r11 = 0x00000151cfe7dbd0
    r12 = 0x0000000000000011   r13 = 0x000000000000005b
    r14 = 0x0000000000000000   r15 = 0x00000000000001ce
    rip = 0x00007fffda3075f8
    Found by: given as instruction pointer in context
 1  chrome.dll!Filter::matches(char const *,int,FilterOption,char const *,BloomFilter *,char const *,int) [filter.cc : 550 + 0xb]
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff0d0
    rip = 0x00007fffda30777e
    Found by: stack scanning
 2  win32u.dll + 0x1284
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff108
    rip = 0x00007ff81b0e1284
    Found by: stack scanning
 3  guard64.dll + 0x175e0
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff110
    rip = 0x00007fffffe175e0
    Found by: stack scanning
 4  chrome.dll!AdBlockClient::hasMatchingFilters(Filter *,int,char const *,int,FilterOption,char const *,BloomFilter *,char const *,int,Filter * *) [ad_block_client.cc : 626 + 0x32]
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff160
    rip = 0x00007fffda3030a6
    Found by: stack scanning
 5  guard64.dll + 0x5d23
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff170
    rip = 0x00007fffffe05d23
    Found by: stack scanning
 6  chrome.dll!isNoFingerprintDomainHashSetMiss(HashSet *,char const *,int) [ad_block_client.cc : 694 + 0xd]
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff1d0
    rip = 0x00007fffda30345f
    Found by: stack scanning
 7  chrome.dll!operator new(unsigned __int64) [new_scalar.cpp : 35 + 0x8]
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff1e0
    rip = 0x00007fffdc0d6fff
    Found by: stack scanning
 8  chrome.dll!AdBlockClient::matches(char const *,FilterOption,char const *,Filter * *,Filter * *) [ad_block_client.cc : 804 + 0x54]
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff250
    rip = 0x00007fffda303af0
    Found by: stack scanning
 9  chrome.dll!base::MessagePumpForUI::ScheduleWork() [message_pump_win.cc : 106 + 0x11]
    rbp = 0x000000382bdff5a8   rsp = 0x000000382bdff260
    rip = 0x00007fffd91ff328
    Found by: stack scanning
```

**Comment by simonhong** [comment]
Recently, c++ adblock lib is replaced with rust based adblock lib - https://github.com/brave/brave-core/pull/2569
So, let's wait and see this happens again with new rust libs.

**Comment by Binklebonk** [comment]
I've been using the Nightly build (0.69.44) for a few days now.  This issue is resolved for me.  Nightly is actually more stable than Release for me at the moment because of this issue.

**Comment by bsclifton** [comment]
Thanks for confirming, @Binklebonk! 😄 

@Helldembez @chrisknepper - if you're still using Brave and the crash is fairly easy to trigger, can you please try either Nightly or Dev channel and report back?
https://brave.com/download-nightly/
https://brave.com/download-dev/

**Comment by Helldembez** [comment]
> 
> 
> Thanks for confirming, @Binklebonk! 😄
> 
> @Helldembez @chrisknepper - if you're still using Brave and the crash is fairly easy to trigger, can you please try either Nightly or Dev channel and report back?
> https://brave.com/download-nightly/
> https://brave.com/download-dev/

I will test it when I have the time. Will report back with the result.

**Comment by Helldembez** [comment]
Tested and found fixed in Version 0.68.100 Chromium: 75.0.3770.100 (Official Build) dev (64-bit)
